### PR TITLE
Oracle module basic functioning

### PR DESCRIPTION
**Pull request by musnit**
Add initial flow for transactions for creating claims

### CODE REVIEW

File: x/oracle/client/cli/query.go [hunk 1]
```diff
@@ -1,26 +1,38 @@
...
+	"strconv"
+
+	"github.com/swishlabsco/cosmos-ethereum-bridge/x/oracle"
+
+	"github.com/cosmos/cosmos-sdk/client/context"
+	"github.com/cosmos/cosmos-sdk/codec"
+	"github.com/spf13/cobra"
+)
+
+// GetCmdGetProphecy queries information about a name
```
Comment (denalimarsh) [inline x/oracle/client/cli/query.go:14]: Change comment to reflect code functionality

File: x/oracle/client/rest/rest.go [hunk 2]
```diff
@@ -1,32 +1,41 @@
...
 // RegisterRoutes - Central function to define routes that get registered by the main application
 func RegisterRoutes(cliCtx context.CLIContext, r *mux.Router, cdc *codec.Codec, storeName string) {
-	// r.HandleFunc(fmt.Sprintf("/%s/names", storeName), buyNameHandler(cdc, cliCtx)).Methods("POST")
-	// r.HandleFunc(fmt.Sprintf("/%s/names/{%s}", storeName, restName), resolveNameHandler(cdc, cliCtx, storeName)).Methods("GET")
+	r.HandleFunc(fmt.Sprintf("/%s/prophecies", storeName), makeClaimHandler(cdc, cliCtx)).Methods("POST")
+	r.HandleFunc(fmt.Sprintf("/%s/prophecies/{%s}", storeName, restNonce), getProphecyHandler(cdc, cliCtx, storeName)).Methods("GET")
 }
 
-/*
 type buyNameReq struct {
```
Comment (denalimarsh) [inline x/oracle/client/rest/rest.go:29]: Update struct name `buyNameReq` to something more appropriate

File: x/oracle/errors.go [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+// Local code type
+type CodeType = sdk.CodeType
+
+//Exported code type numbers
+const (
+	DefaultCodespace sdk.CodespaceType = "oracle"
+
+	CodeInvalidNonce  CodeType = 1
+	CodeNotFound      CodeType = 2
+	CodeMinimumTooLow CodeType = 3
```
Comment (denalimarsh) [inline x/oracle/errors.go:16]: Could we update the `CodeMinimumTooLow` variable name to be more reflective of the error? Could lead to misunderstanding of the expected error, as the transaction value could be too low, or the validator's available coins could be too low, etc.

File: x/oracle/handler.go [hunk 4]
```diff
@@ -21,6 +22,42 @@ func NewHandler(keeper Keeper) sdk.Handler {
...
+	newProphecy := NewBridgeProphecy(msg.Nonce, PendingStatus, getMinimumClaims(), bridgeClaims)
+	err := keeper.createProphecy(ctx, newProphecy)
+	if err != nil {
+		return err.Result()
+	}
+	return sdk.Result{}
+}
+
+func getMinimumClaims() int {
+	minimumClaims := float64(getTotalNumberValidators()) * DefaultConsensusNeeded
```
Comment (denalimarsh) [inline x/oracle/handler.go:55]: `minimumClaims` should be the minimum power needed, not the minimum number of validators.

File: x/oracle/handler.go [hunk 5]
```diff
@@ -21,6 +22,42 @@ func NewHandler(keeper Keeper) sdk.Handler {
...
+	bridgeClaims := []BridgeClaim{bridgeClaim}
+	newProphecy := NewBridgeProphecy(msg.Nonce, PendingStatus, getMinimumClaims(), bridgeClaims)
+	err := keeper.createProphecy(ctx, newProphecy)
+	if err != nil {
+		return err.Result()
+	}
+	return sdk.Result{}
+}
+
+func getMinimumClaims() int {
```
Comment (denalimarsh) [inline x/oracle/handler.go:54]: I think `getPowerThreshold` is a more appropriate description of what this helper function should accomplish.

File: x/oracle/msgs.go [hunk 6]
```diff
@@ -8,16 +8,20 @@ import (
 
 // MsgMakeBridgeClaim defines a SetName message
```
Comment (denalimarsh) [inline x/oracle/msgs.go:9]: `SetName` message is outdated.

File: x/oracle/types.go [hunk 7]
```diff
@@ -15,33 +18,49 @@ type BridgeClaim struct {
...
 		Amount:         amount,
 	}
 }
 
 // BridgeProphecy is a struct that contains all the metadata of an oracle ritual
 type BridgeProphecy struct {
-	Status        string        `json:"status"`
 	Nonce         int           `json:"nonce"`
+	Status        string        `json:"status"`
 	MinimumClaims int           `json:"minimum_claims"` //The minimum number of claims needed before completion logic is checked
```
Comment (denalimarsh) [inline x/oracle/types.go:40]: Updating `MinimumClaims` to `MinimumPower` in the `BridgeProphecy` struct makes sense, as many parts of the application interact with the data structure.

**Review by denalimarsh — CHANGES_REQUESTED** [review]

**Comment by denalimarsh** [comment]
Requested changes are primarily centered on minimum validator count being updated to minimum power threshold.

**Comment by musnit** [comment]
All suggestions fixed in new PR https://github.com/swishlabsco/cosmos-ethereum-bridge/pull/4

Due to the refactor, let's merge this and let the fixes be included in that PR.

**Pull request merged**